akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -380,11 +382,12 @@ object StreamLayout {
 
     override def toString =
       s"""
-        | Module: ${this.attributes.nameOrDefault("unnamed")}
-        | Modules: ${subModules.iterator.map(m ⇒ "\n   " + m.attributes.nameOrDefault(m.getClass.getName)).mkString("")}
-        | Downstreams: ${downstreams.iterator.map { case (in, out) ⇒ s"\n   $in -> $out" }.mkString("")}
-        | Upstreams: ${upstreams.iterator.map { case (out, in) ⇒ s"\n   $out -> $in" }.mkString("")}
-        |""".stripMargin
+         |  Module: ${this.attributes.nameOrDefault("unnamed")}
+         |  Modules:
+         |    ${subModules.iterator.map(m ⇒ m.attributes.nameLifted.getOrElse(m.toString.replaceAll("\n", "\n    "))).mkString("\n    ")}
```

Comment:
Also note that it doesn't fall back to the modules' class' name, but to its String representation.